app/scripts/modules/core/cluster/filter/clusterFilter.service.ts

```diff
@@ -46,7 +47,18 @@ export class ClusterFilterService {
 
   public groupsUpdatedStream: Subject<IClusterGroup[]> = new Subject<IClusterGroup[]>();
 
-  public updateClusterGroups = debounce((application?: Application) => {
+  private lastApplication: Application;
+
+  private isFilterable: (sortFilter: any) => boolean = this.filterModelService.isFilterable;
+
+  public constructor(private ClusterFilterModel: any, private MultiselectModel: any,
+                     private $log: ILogService, private $stateParams: StateParams,
+                     private filterModelService: any) {
+    'ngInject';
+  }
+
+  @Debounce(25)
```

Comment:
Agreed, much nicer. But in this case I'm not sure a function that returns a value should be debounced.